src/modules/mavlink/mavlink_receiver.h

```diff
@@ -189,90 +188,101 @@ class MavlinkReceiver
-	param_t _p_flow_rot;
-	param_t _p_flow_maxr;
-	param_t _p_flow_minhgt;
-	param_t _p_flow_maxhgt;
+	Mavlink *_mavlink;
+	MavlinkFTP               _mavlink_ftp;
+	MavlinkLogHandler        _mavlink_log_handler;
+	MavlinkTimesync          _mavlink_timesync;
+	MavlinkMissionManager    _mission_manager;
+	MavlinkParametersManager _parameters_manager;
+
+	mavlink_status_t _status {}; ///< receiver status, used for mavlink_parse_char()
+	struct vehicle_attitude_s _att {};
+	struct vehicle_local_position_s _hil_local_pos {};
+	struct vehicle_land_detected_s _hil_land_detector {};
+	struct vehicle_control_mode_s _control_mode {};
+
+	orb_advert_t _actuator_controls_pubs[4] = {nullptr, nullptr, nullptr, nullptr};
+
+	orb_advert_t _accel_pub{nullptr};
+	orb_advert_t _airspeed_pub{nullptr};
+	orb_advert_t _attitude_pub{nullptr};
+	orb_advert_t _att_sp_pub{nullptr};
+	orb_advert_t _baro_pub{nullptr};
+	orb_advert_t _battery_pub{nullptr};
+	orb_advert_t _cmd_pub{nullptr};
+	orb_advert_t _command_ack_pub{nullptr};
+	orb_advert_t _collision_report_pub{nullptr};
+	orb_advert_t _debug_array_pub{nullptr};
+	orb_advert_t _debug_key_value_pub{nullptr};
+	orb_advert_t _debug_value_pub{nullptr};
+	orb_advert_t _debug_vect_pub{nullptr};
+	orb_advert_t _distance_sensor_pub{nullptr};
+	orb_advert_t _flow_distance_sensor_pub{nullptr};
+	orb_advert_t _flow_pub{nullptr};
+	orb_advert_t _follow_target_pub{nullptr};
+	orb_advert_t _global_pos_pub{nullptr};
+	orb_advert_t _gps_inject_data_pub{nullptr};
+	orb_advert_t _gps_pub{nullptr};
+	orb_advert_t _gyro_pub{nullptr};
+	orb_advert_t _hil_distance_sensor_pub{nullptr};
+	orb_advert_t _land_detector_pub{nullptr};
+	orb_advert_t _landing_target_pose_pub{nullptr};
+	orb_advert_t _local_pos_pub{nullptr};
+	orb_advert_t _mag_pub{nullptr};
+	orb_advert_t _manual_pub{nullptr};
+	orb_advert_t _mocap_odometry_pub{nullptr};
+	orb_advert_t _obstacle_distance_pub{nullptr};
+	orb_advert_t _offboard_control_mode_pub{nullptr};
+	orb_advert_t _ping_pub{nullptr};
+	orb_advert_t _pos_sp_triplet_pub{nullptr};
+	orb_advert_t _radio_status_pub{nullptr};
+	orb_advert_t _rc_pub{nullptr};
+	orb_advert_t _rates_sp_pub{nullptr};
+	orb_advert_t _transponder_report_pub{nullptr};
+	orb_advert_t _trajectory_waypoint_pub{nullptr};
+	orb_advert_t _tune_control_sub{nullptr};
+	orb_advert_t _visual_odometry_pub{nullptr};
+
+
+	static constexpr unsigned MOM_SWITCH_COUNT{8};
+
+	static const int _gps_inject_data_queue_size{6};
+
+	int _actuator_armed_sub{-1};
+	int _control_mode_sub{-1};
+	int _vehicle_attitude_sub{-1};
+
+	int _hil_frames{0};
+	int _orb_class_instance{-1};
+
+	uint64_t _global_ref_timestamp{0};
+	uint64_t _old_timestamp{0};
+
+	bool _hil_local_proj_inited{false};
+
+	float _hil_local_alt0{0.0f};
+
+	struct map_projection_reference_s _hil_local_proj_ref {};
+	struct offboard_control_mode_s _offboard_control_mode {};
+
+	uint8_t _mom_switch_pos[MOM_SWITCH_COUNT] {};
+	uint16_t _mom_switch_state{0};
+
+	param_t _p_bat_crit_thr{0};
```

Comment:
Similar to the subscription, 0 is actually a valid param it.
```suggestion
	param_t _p_bat_crit_thr{PARAM_INVALID};
```

Reply:
Fixed. Thanks!

Reply:
@dagar , I've changed them to match the old Constructor's initializer list:

```
	param_t _p_bat_crit_thr{param_find("BAT_CRIT_THR")};
	param_t _p_bat_emergen_thr{param_find("BAT_EMERGEN_THR")};
	param_t _p_bat_low_thr{param_find("BAT_LOW_THR")};
	param_t _p_flow_maxhgt{param_find("SENS_FLOW_MAXHGT")};
	param_t _p_flow_maxr{param_find("SENS_FLOW_MAXR")};
	param_t _p_flow_minhgt{param_find("SENS_FLOW_MINHGT")};
	param_t _p_flow_rot{param_find("SENS_FLOW_ROT")};
```

I didn't realize these calls were possible at the declarations before now. If I'm breaking something by doing this please let me know.

Reply:
Hardware behavior checks out with QGC 3.4.4 and this change.